### DISCUSSION TITLE
Bot usability status

### DISCUSSION THREAD

**Discussion by jguillen-lab** [Skylar-Tech/node-red-contrib-matrix-chat#126] [General]
Hi. Thanks for your work. So far, I've tried everything (using Pantalaimon and creating a specific user for Node-RED). The truth is, I haven't had any success with either option sending messages. I understand that currently nodes cannot be used if you are using E2EE as indicated in #124 and #118.

**Comment by skylord123** [comment]
Yeah we are working on adding device verification which would allow e2ee keys to be shared fixing this problem. This is being tracked under https://github.com/Skylar-Tech/node-red-contrib-matrix-chat/issues/124

I have a branch that device verification is almost working there is just an error that is thrown when trying to send the keys to the server. This is an area we could use help. The docs for matrix-js-sdk are sparse and there aren't really any examples of how to do what we want to accomplish except for the matrix-react-sdk. Problem is I am not a react dev so I struggle a bit trying to figure out what missing piece we need to implement. 

As for Pantalaimon I think it requires a setting in matrix-js-sdk to be set to actually function properly. They have a flag you need to set to get external e2ee handlers to work (I found this recently). I'll have to look into implementing this as an option in the server config node in Node-RED (something like a checkbox that you enable when using Pantalaimon).

Currently I use this module for notifications. Since I host my own server I just leave the rooms unencrypted as the messages will never leave my server anyways. I definitely understand the need to have fully functional e2ee (with device verification) for using servers out of your control (such as the public matrix one) or having the messages encrypted in case the server is hacked.

If anyone wants to collaborate on this or help with getting device verification working feel free to check out my issue I created for tracking this. There is also the matrix chat channel  https://matrix.to/#/#node-red-contrib-matrix-chat:skylar.tech that anyone is welcome to join and discuss this.

**Reply by jguillen-lab** [reply]
Thank you very much for your detailed response. As soon as I have an egg, I will try with the dev branch and the sample flow. I will keep you informed of the progress here.
